cms/djangoapps/contentstore/tasks.py

```diff
@@ -47,10 +51,243 @@
+            all_lang_transcripts.update({'en': video.sub})
+        for lang, _ in all_lang_transcripts.items():
+            transcript_already_present = is_transcript_available(
+                clean_video_id(video.edx_video_id),
+                lang
+            )
+            if transcript_already_present and force_update:
+                sub_tasks.append(async_migrate_transcript_subtask.s(
+                    video, lang, True, **kwargs
+                ))
+            elif not transcript_already_present:
+                sub_tasks.append(async_migrate_transcript_subtask.s(
+                    video, lang, False, **kwargs
+                ))
+    LOGGER.info("[Transcript migration] Migrating %s transcripts", len(sub_tasks))
+    callback = task_status_callback.s()
+    status = chord(sub_tasks)(callback)
+    LOGGER.info(
+        "[Transcript migration] process for course %s ended. Processed %s transcripts",
+        course_key,
+        len(status.get())
+    )
+    return status.get()
+
+
+def get_videos_from_store(course_key):
+    """
+    Returns all videos in a course as list.
+
+    Arguments:
+        course_key: CourseKey object
+    """
+    store = modulestore()
+    all_videos = []
+    for video in store.get_items(course_key, qualifiers={'category': 'video'},
+                                 revision=ModuleStoreEnum.RevisionOption.published_only, include_orphans=False):
+        all_videos.append(video)
+
+    for video in store.get_items(course_key, qualifiers={'category': 'video'},
+                                 revision=ModuleStoreEnum.RevisionOption.draft_only, include_orphans=False):
+        all_videos.append(video)
+
+    return all_videos
+
+
+@chord_task(
+    bind=True,
+    base=LoggedPersistOnFailureTask,
+    default_retry_delay=RETRY_DELAY_SECONDS,
+    max_retries=2,
+    time_limit=VIDEO_LEVEL_TIMEOUT_SECONDS
+)
+def async_migrate_transcript_subtask(self, *args, **kwargs):
+    #pylint: disable=unused-argument
+    """
+    Migrates a transcript of a given video in a course as a new celery task.
+    """
+    video, language_code, force_update = args
+    commit = kwargs['commit']
+    result = None
+    if commit is not True:
+        return 'Language {0} transcript of video {1} will be migrated'.format(
+            language_code,
+            video.edx_video_id
+        )
+    LOGGER.info("[Transcript migration] process for %s transcript started", language_code)
+    try:
+        transcript_info = video.get_transcripts_info()
+        transcript_content, _, _ = get_transcript_from_contentstore(
+            video, language_code, Transcript.SJSON, transcript_info)
+
+        if not clean_video_id(video.edx_video_id):
+            video.edx_video_id = create_external_video('external-video')
+            video.save_with_metadata(user=User.objects.get(username='staff'))
+        if video.edx_video_id:
+            result = save_transcript_to_storage(
+                clean_video_id(video.edx_video_id),
```

Comment:
I think we can do `clean_video_id` only once like this.

```python
edx_video_id = clean_video_id(video.edx_video_id)

if not edx_video_id:
    video.edx_video_id = create_external_video('external-video')
    video.save_with_metadata(user=User.objects.get(username='staff'))

if edx_video_id:
    result = save_transcript_to_storage(
        edx_video_id,
        language_code,
        transcript_content,
        Transcript.SJSON,
        force_update
   )
```

And we need to remove `clean_video_id` from `save_transcript_to_storage` as we not making any decision on `edx_video_id` in it.